worlds/banjo_tooie/Options.py

```diff
@@ -0,0 +1,513 @@
+    range_start = 0
+    range_end = 100
+    default = 40
+
+class TripTrapWeight(Range):
+    """The weight of Trip Traps in the trap pool.
+    Requires Traps to have an effect"""
+    display_name = "Trip Trap Weight"
+    range_start = 0
+    range_end = 100
+    default = 40
+    
+class SlipTrapWeight(Range):
+    """The weight of Slip Traps in the trap pool.
+    Requires Traps to have an effect"""
+    display_name = "Slip Trap Weight"
+    range_start = 0
+    range_end = 100
+    default = 40
+    
+class TransformTrapWeight(Range):
+    """The weight of Transform Traps in the trap pool.
+    Requires Traps to have an effect"""
+    display_name = "Transform Trap Weight"
+    range_start = 0
+    range_end = 100
+    default = 40
+    
+class SquishTrapWeight(Range):
+    """The weight of Squish Traps in the trap pool.
+    Requires Traps to have an effect"""
+    display_name = "Squish Trap Weight"
+    range_start = 0
+    range_end = 100
+    default = 20
+    
+class KingJingalingHasJiggy(DefaultOnToggle):
+    """King Jingaling will always have a Jiggy for you."""
+    display_name = "King Jingaling Jiggy"
+
+class SkipPuzzles(DefaultOnToggle):
+    """Open world entrances without having to go to Jiggywiggy."""
+    display_name = "Skip Puzzles"
+
+class Backdoors(Toggle):
+    """Opens many one way switches on game start, allowing for more backdoor access to levels.
+    The following gates are preopened: MT>TDL, MT>HFP, GGM>WW, WW>TDL
+    For MT>TDL only the gate accessed from TDL's side is opened and for GGM>WW the boulders are still unexploded
+    The bridge from HFPs entrance is pre-moved to allow secondary access to clifftop
+    George is also pre-dropped to make HFP>JRL more accessible"""
+    display_name = "Open Backdoors"
+
+class OpenHag1(DefaultOnToggle):
+    """HAG 1 boss fight is opened when Cauldron Keep is opened. Only 55 jiggies are needed to win."""
+    display_name = "HAG 1 Open"
+
+class RandomizeWorlds(Toggle):
+    """Worlds will open in a randomized order. Randomized Moves and Puzzle Skip Required."""
+    display_name = "Randomize Worlds"
+
+class RandomizeWorldZones(Toggle):
+    """World Entrances will warp you to a different world. This does not affect Chuffy."""
+    display_name = "Randomize World Entrances"
+
+class RandomizeStopnSwap(Toggle):
+    """Mystery Eggs, their rewards, and the Ice Key are scattered across the MultiWorld."""
+    display_name = "Randomize Stop n Swap"
+
+class SkipToT(Choice):
+    """Choose whether to play the full quiz, start at round 3, or skip it."""
+    display_name = "Tower of Tragedy Quiz"
+    option_full = 0
+    option_skip = 1
+    option_round_3 = 2
+    default = 1
+
+class LogicType(Choice):
+    """Choose your logic difficulty and difficulty of tricks you are expected to perform to reach certain areas.
+    Please be aware that if you plan on randomizing worlds with BK Moves in the pool, you cannot use "intended" logic.""" 
```

Comment:
What does the bit "you cannot use 'intended' logic" mean?

Reply:
The option to select the logic used gives 4 options: intended, easy tricks, hard tricks, glitches. Intended logic is the name for one of them, which sticks to intended strategies only.